src/test/java/com/google/firebase/auth/OidcProviderConfigTest.java

```diff
@@ -157,4 +196,14 @@ public void testUpdateRequestMissingIssuer() {
   public void testUpdateRequestInvalidIssuerUrl() {
     new OidcProviderConfig.UpdateRequest("oidc.provider-id").setIssuer("not a valid url");
   }
+
```

Comment:
What happens if the developer sets responseTypeCode but doesn't specify a client secret? Should the SDK validate that? Let's at least have a test case to exercise that scenario.

Reply:
I'm letting the backend deal with it - they will return an error. I can't do this check in update, and if you want me to make this check in create then we'll be forcing the dev to call setClientSecret before calling setCodeResponseType. If you prefer I do that let me know.

Reply:
See my other comment.